src/react.tsx

```diff
@@ -31,7 +47,19 @@ function normalizeQuery<STORE extends typeof MSTGQLStore.Type, DATA>(
 ): Query<DATA> {
   if (typeof query === "function") return query(store)
   if (query instanceof Query) return query
-  return store.query(query, variables, { fetchPolicy, raw })
+  if (typeof window === "undefined") {
+    const queryPromise = store.query<DATA>(query, variables, {
```

Comment:
Looks like this is duplicated in both conditional branches. I would suggest moving `pushPromise` call to be conditional only.

Reply:
@FredyC how so? I am concered about this though
```js
  if (typeof query === "function") return query(store)
  if (query instanceof Query) return query
```
those cases are not covered and it seems like they should be